File: app/src/main/java/com/lazysoul/kotlinwithandroid/ui/main/MainMvpView.java
package com.lazysoul.kotlinwithandroid.ui.main;

import com.lazysoul.kotlinwithandroid.common.BaseMvpView;
import com.lazysoul.kotlinwithandroid.datas.Todo;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Lazysoul on 2017. 7. 9..
 */

interface MainMvpView extends BaseMvpView {

    void onUpdateTodoList(List<Todo> todoList);

    void onUpdateTodo(Todo todo);

    void onCreatedTodo(Todo todo);

    void showEmtpyView();

    void onCreatedSampes(List<Todo> todoList);
}

File: app/src/main/kotlin/com/lazysoul/kotlinwithandroid/ui/main/MainMvpView.kt
package com.lazysoul.kotlinwithandroid.ui.main

import com.lazysoul.kotlinwithandroid.common.BaseMvpView
import com.lazysoul.kotlinwithandroid.datas.Todo

/**
 * Created by Lazysoul on 2017. 7. 9..
 */

interface MainMvpView : BaseMvpView {

    fun onUpdateTodoList(todoList: List<Todo>)

    fun onCreatedTodo(todo: Todo)

    fun showEmtpyView()

    fun onSuccessCreateSampes()
}
